src/crush/CrushCompiler.cc

```diff
@@ -608,6 +608,10 @@ int CrushCompiler::parse_rule(iter_t const& i)
   }
 
   int ruleset = int_node(i->children[start]);
+  if (ruleset_name.count(ruleset)) {
+    err << "ruleset '" << ruleset << "' already defined in rule '" << ruleset_name[ruleset] <<"'\n"<< std::endl;
+    return -1;
```

Comment:
in `CrushCompiler::parse_crush()`, only the last returned `r` is returned. so in the test case of src/test/cli/crushtool/check-overlapped-rules.t, a partial compiled crush map is generated. if user is using a script to generate the map, he/she might not notice this problem.